static/app/utils/profiling/hooks/useProfileEventsSort.tsx

```diff
@@ -0,0 +1,36 @@
+import {useMemo} from 'react';
+
+import {GridColumnSortBy} from 'sentry/components/gridEditable';
+import {decodeScalar} from 'sentry/utils/queryString';
+import {useLocation} from 'sentry/utils/useLocation';
+
+interface UseProfileEventsSortOptions<F> {
+  allowedKeys: readonly F[];
+  fallback: {
+    key: F;
+    order: 'asc' | 'desc';
+  };
+  key: string;
+}
+
+export function useProfileEventsSort<F extends string>(
```

Comment:
Is this just to get grid options, I'm a little confused on how you'd use this? If it's to compute grid options I'd just use a function instead of a hook here.

Reply:
It's to derive the `GridColumnSort` from the query parameters which will then be passed to either the `useProfileEvents` hook or the `ProfileEventsTable` in the next PR. If we made this a function here, we'd have to write the boiler plate to read it from the query parameters every time, but I don't have strong opinions either way if you prefer that.

Reply:
You could pass location as the argument, I'm saying that because that way you decouple this from hooks and that makes it easier to test. That said, this is too small of a use-case to nit-pick, I'm fine with either :)

Reply:
I've updated to use a function instead. The tests are much cleaner now.